exporter/metric/metric_test.go

```diff
@@ -118,3 +118,98 @@ func TestRecordToMpb(t *testing.T) {
 		t.Errorf("%v", aggError)
 	}
 }
+
+
+func TestResourceToMonitoredResourcepb(t *testing.T) {
+	resList := []*resource.Resource{
```

Comment:
Could you please add a couple of tests cases:

* Only Partial Data provided - e.g.
```
kv.String("cloud.provider", "gcp"),
kv.String("k8s.pod.name", "opentelemetry-pod-autoconf"),
```

* Cloud Provider = GCP, but invalid data / unknown resource type - e.g. 
```
kv.String("cloud.provider", "gcp"),
kv.String("cloud.zone", "us-central1-a"),
```

Reply:
Actually maybe those are both the same case 😆 

Does Cloud Monitoring distinguish between mandatory and optional fields? Do we need to do something in our code to validate that? i.e. if we don't have all the fields for a given resource, should we fallback to global or something? Otherwise Cloud Monitoring may reject the request entirely? (I could be wrong - it wasn't obvious to me what happens from the API docs)

Reply:
Sure, but my question is what will happen if we attempt to export data that has the following attributes?

```go
kv.String("cloud.provider", "gcp"),
kv.String("cloud.zone", "us-central1-a"),
kv.String("k8s.cluster.name", "opentelemetry-cluster"),
```

Your mapping code will identify it as type "k8s_container" due to including a "k8s.*" attribute, but it doesn't have all the required (?) attributes. What will happen in this case? (and we should have a test case to cover it)

Reply:
I agree. Initially, I had the same opinion and treated types of missing fields as global monitored resources in the same way of Go OpenCensus (https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/blob/master/resource.go#L162). Now I would strongly recommend keeping consistent with Opencensus since the backend would not receive monitored resources with missing fields. @dukeng

Reply:
Okay cool - looks good now.

I think you could have one more test case for a "global" resource type, e.g.:
```go
kv.String("cloud.provider", "gcp"),
kv.String("cloud.zone", "us-central1-a"),
```